File: app/src/main/java/jp/java_conf/ikastyle/Common/Const/GearPowerPositionKind.java
package jp.java_conf.ikastyle.Common.Const;

public enum GearPowerPositionKind {
    SUB(0), // サブギアは0
    HEAD(1), // アタマのメインは1
    CLOTHING(2), // フクのメインは2
    SHOES(3); // クツのメインは3

    private final int id;

    GearPowerPositionKind(int id){
        this.id = id;
    }

    public int getId(){
        return this.id;
    }

    // idを受け取って該当idのGearKindを返す
    public static GearPowerPositionKind getGearPowerPositionKind(int id){
        GearPowerPositionKind[] gearPowerPositionKinds = GearPowerPositionKind.values();
        for (GearPowerPositionKind item: gearPowerPositionKinds) {
            if(item.getId() == id){
                return item;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/splatool/ikastyle/common/const/GearPowerPositionKind.kt
package com.splatool.ikastyle.common.const

/*
 * ギアパワーの場所種別(サブ,アタマメイン,フクメイン,クツメイン)のEnum
 */
enum class GearPowerPositionKind(  // クツのメインは3
    private val id: Int
) {
    SUB(0),  // サブギアは0
    HEAD(1),  // アタマのメインは1
    CLOTHING(2),  // フクのメインは2
    SHOES(3);

    fun getId(): Int {
        return id
    }

    companion object {
        // idを受け取って該当idのGearPowerPositionKindを返す
        fun getGearPowerPositionKind(id: Int): GearPowerPositionKind? {
            for(item: GearPowerPositionKind in values()){
                if (item.getId() == id) {
                    return item
                }
            }
            return null
        }
    }
}